### DISCUSSION TITLE
Using custom scripts in custom links

### DISCUSSION THREAD

**Discussion by zshorvat** [netbox-community/netbox#7956] [Q&A] [answered]
We would like to be able to call _custom scripts_ from _custom links_, but there seems to be a complication: while the _custom script_ is accessible as an API call, the example suggest to use a POST call, constructing all data, token, etc which might not even be feasible in GET, that I assume I could use from _custom links_.

Is there a way to still achieve this somehow?

Thanks for the answers in advance!

**Comment by drmsoffall** [comment, marked as answer]
I'm not sure about directly executing a script via GET, but you _can_ pre-populate a custom script form by specifying the field values in the URI.  For example, if your script had an ObjectVar called `mydev` and the link was on a device, you could provide the pk by appending `?mydev={{ obj.id }}` to the script URI when building the link.  This would pre-select that object, but the user would still have to click the Run button to execute it.

**Reply by zshorvat** [reply]
Thanks a lot, I think that will work after we upgrade (we use multiobjectvars in places that used to have a kind of bug).
